### ISSUE TITLE
System property tests fail on NI-XNET 15.5

### ISSUE THREAD

**Issue by epage** [ni/nixnet-python#156]
Our integration tests were tested out on 15.5 an ran into this problem.

- Python version: Python 3.4
- NI-XNET version: NI-XNET 15.5

```python3
___________________________ test_device_properties ____________________________

nixnet_in_interface = 'CAN1'

    @pytest.mark.integration
    def test_device_properties(nixnet_in_interface):
        """Verify Device properties.
    
        Ideally we'd match these against a known piece of hardware to ensure the
        correct values are being returned.  That makes it hard though for someone
        to run these tests with any piece of hardware for most properties.  For
        now, we'll just verify the calls don't call catastrophically and someone
        can always run py.test with ``-s``_.  For ``int_refs_all``, we can at least
        make sure that one device reports ``nixnet_in_interface``_.
        """
        with system.System() as sys:
            devs = list(sys.dev_refs)
            assert 0 < len(devs), "Pre-requisite failed"
            for dev in devs:
                in_intfs = [intf for intf in dev.intf_refs_all if intf == nixnet_in_interface]
                if len(in_intfs) == 1:
                    break
            else:
>               raise RuntimeError("Pre-requisite failed: can't find device for interface")
E               RuntimeError: Pre-requisite failed: can't find device for interface

tests\test_system.py:121: RuntimeError
_____________________________ test_intf_container _____________________________

nixnet_in_interface = 'CAN1'

    @pytest.mark.integration
    def test_intf_container(nixnet_in_interface):
        with system.System() as sys:
            intfs = list(sys.intf_refs_all)
            in_intfs = [intf for intf in intfs if intf == nixnet_in_interface]
>           assert len(in_intfs) == 1
E           assert 0 == 1
E            +  where 0 = len([])

tests\test_system.py:140: AssertionError
_______________________________ test_intf_blink _______________________________

nixnet_in_interface = 'CAN1'

    @pytest.mark.integration
    def test_intf_blink(nixnet_in_interface):
        """Verify LEDs can be blinked.
    
        Since we don't have a camera watching the LEDs, we can at least test if we
        are having an impact by checking
    
        - `ENABLE` works anytime except a session is in use
        - `DISABLE` always works
    
        Sleeps are in place in case this in case a visual confirmation is possible
        by the developer.
        """
        with system.System() as sys:
            intfs = list(sys.intf_refs_all)
            in_intfs = [intf for intf in intfs if intf == nixnet_in_interface]
>           assert len(in_intfs) == 1, "Pre-requisite failed"
E           AssertionError: Pre-requisite failed
E           assert 0 == 1
E            +  where 0 = len([])

tests\test_system.py:175: AssertionError
____________________________ test_intf_properties _____________________________

nixnet_in_interface = 'CAN1'

    @pytest.mark.integration
    def test_intf_properties(nixnet_in_interface):
        """Verify Interface properties.
    
        Ideally we'd match these against a known piece of hardware to ensure the
        correct values are being returned.  That makes it hard though for someone
        to run these tests with any piece of hardware.  For now, we'll just verify
        the calls don't call catastrophically and someone can always run py.test
        with ``-s``_.
        """
        with system.System() as sys:
            intfs = list(sys.intf_refs_all)
            in_intfs = [intf for intf in intfs if intf == nixnet_in_interface]
>           assert len(in_intfs) == 1, "Pre-requisite failed"
E           AssertionError: Pre-requisite failed
E           assert 0 == 1
E            +  where 0 = len([])

tests\test_system.py:207: AssertionError
```

**Comment by epage** [comment]
My theory is that the `intf_refs_all` versions of the properties is fairly new and doesn't work on the older versions.  But if so, why do the calls return no interfaces rather than throwing an error?

**Comment by epage** [comment]
Alternatively, this was some kind of hardware failure.

tox runs the same tests on multiple versions of python.  The first run only had these failures.  The other runs failed as if there wasn't any hardware in the system.  These tests that failed were at the tail end of the run, so that might have been when the hardware first went on the fritz.

```
nixnet/_frames.py::nixnet._frames._calculate_payload_size PASSED
nixnet/_frames.py::nixnet._frames._calculate_payload_unit_size PASSED
nixnet/_frames.py::nixnet._frames._get_frame_payload_length PASSED
nixnet/_frames.py::nixnet._frames._split_payload_length PASSED
nixnet/_utils.py::nixnet._utils.flatten_items PASSED
nixnet/types.py::nixnet.types.CanFrame.__repr__ PASSED
nixnet/types.py::nixnet.types.CanFrame.from_raw PASSED
nixnet/types.py::nixnet.types.CanFrame.to_raw PASSED
nixnet/types.py::nixnet.types.CanIdentifier.__int__ PASSED
nixnet/types.py::nixnet.types.CanIdentifier.__repr__ PASSED
nixnet/types.py::nixnet.types.CanIdentifier.from_raw PASSED
nixnet/types.py::nixnet.types.RawFrame.__repr__ PASSED
tests/test_errors.py::test_success PASSED
tests/test_errors.py::test_known_error PASSED
tests/test_errors.py::test_unknown_error PASSED
tests/test_errors.py::test_known_warning PASSED
tests/test_errors.py::test_unknown_warning PASSED
tests/test_errors.py::test_driver_call PASSED
tests/test_examples.py::test_can_frame_queued_empty_session[input_values0] PASSED
tests/test_examples.py::test_can_frame_queued_empty_session[input_values1] PASSED
tests/test_examples.py::test_can_frame_queued_empty_session[input_values2] PASSED
tests/test_examples.py::test_can_frame_queued_empty_session[input_values3] PASSED
tests/test_examples.py::test_can_frame_queued_empty_session[input_values4] PASSED
tests/test_examples.py::test_can_frame_stream_empty_session[input_values0] PASSED
tests/test_examples.py::test_can_frame_stream_empty_session[input_values1] PASSED
tests/test_examples.py::test_can_frame_stream_empty_session[input_values2] PASSED
tests/test_examples.py::test_can_frame_stream_empty_session[input_values3] PASSED
tests/test_examples.py::test_can_frame_stream_empty_session[input_values4] PASSED
tests/test_examples.py::test_can_signal_single_point_empty_session[input_values0] PASSED
tests/test_examples.py::test_can_signal_single_point_empty_session[input_values1] PASSED
tests/test_examples.py::test_can_signal_single_point_empty_session[input_values2] PASSED
tests/test_examples.py::test_can_signal_single_point_empty_session[input_values3] PASSED
tests/test_examples.py::test_can_signal_single_point_empty_session[input_values4] PASSED
tests/test_examples.py::test_can_signal_single_point_empty_session[input_values5] PASSED
tests/test_examples.py::test_can_signal_single_point_empty_session[input_values6] PASSED
tests/test_frames.py::test_iterate_frames_with_empty_payload PASSED
tests/test_frames.py::test_iterate_frames_with_base_payload PASSED
tests/test_frames.py::test_iterate_frames_with_partial_base_payload PASSED
tests/test_frames.py::test_iterate_frames_with_multiple_frames PASSED
tests/test_frames.py::test_iterate_frames_corrupted_frame PASSED
tests/test_frames.py::test_can_identifier_equality PASSED
tests/test_frames.py::test_can_identifier_overflow PASSED
tests/test_frames.py::test_can_identifier_extended_overflow PASSED
tests/test_frames.py::test_raw_frame_equality PASSED
tests/test_frames.py::test_raw_frame_conversion PASSED
tests/test_frames.py::test_can_frame_equality PASSED
tests/test_frames.py::test_serialize_frame_with_empty_payload PASSED
tests/test_frames.py::test_serialize_frame_with_base_payload PASSED
tests/test_frames.py::test_serialize_frame_with_payload_unit PASSED
tests/test_frames.py::test_serialize_frame_with_excessive_payload PASSED
tests/test_frames.py::test_stream_loopback PASSED
tests/test_frames.py::test_queued_loopback PASSED
tests/test_frames.py::test_singlepoint_loopback PASSED
tests/test_frames.py::test_session_frames_container PASSED
tests/test_frames.py::test_session_frames_properties PASSED
tests/test_frames.py::test_session_frame_container PASSED
tests/test_lib.py::test_unsupported_platform PASSED
tests/test_lib.py::test_function_not_supported PASSED
tests/test_session.py::test_flatten_items_invalid PASSED
tests/test_session.py::test_session_container PASSED
tests/test_session.py::test_session_properties PASSED
tests/test_session.py::test_session_properties_transition PASSED
tests/test_session.py::test_parse_can_comm_bitfield PASSED
tests/test_session.py::test_start_explicit PASSED
tests/test_session.py::test_flush_baseline PASSED
tests/test_session.py::test_flush_output_queue PASSED
tests/test_session.py::test_wait_for_intf_communicating PASSED
tests/test_session.py::test_wait_for_transmit_complete PASSED
tests/test_session.py::test_wait_for_intf_remote_wakeup PASSED
tests/test_session.py::test_connect_terminals_failures PASSED
tests/test_session.py::test_disconnect_terminals_failures PASSED
tests/test_session_intf.py::test_intf_container PASSED
tests/test_session_intf.py::test_intf_properties_settable_only_when_stopped PASSED
tests/test_session_intf.py::test_stream_session_requires_baud_rate PASSED
tests/test_session_intf.py::test_sleep_transition PASSED
tests/test_signals.py::test_singlepoint_loopback PASSED
tests/test_signals.py::test_signals_container PASSED
tests/test_system.py::test_system_container PASSED
tests/test_system.py::test_system_properties PASSED
tests/test_system.py::test_system_intf_refs_superset PASSED
tests/test_system.py::test_system_intf_refs_all_superset PASSED
tests/test_system.py::test_device_container PASSED
tests/test_system.py::test_device_properties FAILED
tests/test_system.py::test_intf_container FAILED
tests/test_system.py::test_intf_blink FAILED
tests/test_system.py::test_intf_properties FAILED
```

**Comment by lilesj** [comment]
It appears that "tests/test_system.py::test_device_properties FAILED" causes the hardware to have problems. I am unable to self test the hardware successfully after tox has completed.

**Comment by epage** [comment]
What steps did you take to root cause the problem to that test?

**Comment by lilesj** [comment]
I bid a compare between the Ethernet cDAQ and USB logs. The Ethernet log does have some additional failures we can likely attribute to connection troubles but both logs start failing at the tests/test_system.py::test_device_properties test and continue to fail all tests that use hardware references from that point forward.

Also, the hardware was not in a usable state upon the completion of the test so the problem likely started at that test or soon after.

**Comment by epage** [comment]
So Ajay just ran this and it started failing at "test_system_intf_refs_superset".

One thing to be careful of is that the test that first fails might not be the test that put the machine into a bad state.

**Comment by epage** [comment]
Ajay has done 2 runs of tox without `test_system.py` without problems.

**Comment by epage** [comment]
Fixed by #163
